### PR TITLE
Revamp scons build - Modernize packaging and purge obsolete build logic

### LINKED ISSUE #3452: move all the metadata from setup.py to setup.cfg
https://github.com/gvalkov/setuptools-py2cfg should be helpful. setup.cfg allows to read some file too, using `file: `.

### PR DESCRIPTION
**Pull request by bdbaddog**
Addresses:
Issue #3452 
Issue #3451 
Issue #3450 
Issue #3598 

The logic to generate API docs has been disabled and needs to replaced with logic to build API docs via sphinx.

Move a lot of logic previously in SConstruct to site_scons/
Purge more py2 code.

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `src/CHANGES.txt` (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: runtest.py [hunk 1]
```diff
@@ -478,90 +478,49 @@ def footer(self, f):
...
-    scons_runtest_dir = builddir
+sd = None
+tools_dir = None
+ld = None
+
+if not baseline or baseline == '.':
+    base = cwd
+elif baseline == '-':
+    url = None
+    with os.popen("svn info 2>&1", "r") as p:
```
Comment (mwichmann) [inline runtest.py]: Obsolete?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,66 @@
+[metadata]
+name = SCons
+version=3.9.9a991
+license = MIT
+author = William Deegan
+author_email =bill@baddogconsulting.com
+long_description = file: README.rst
+description = Open Source next-generation build tool.
```
Comment (mwichmann) [inline setup.cfg:8]: Maybe a better description is due?  "next-generation" from a 20-year old? Besides which it doesn't really convey anything.

File: test/packaging/rpm/explicit-target.py [hunk 3]
```diff
@@ -77,7 +77,7 @@
 
 expect = """
 scons: *** Setting target is not supported for rpm.
-""" + test.python_file_line(test.workpath('SConstruct'), 23)
+""" + test.python_file_line(test.workpath('SConstruct'), 12)
```
Comment (mwichmann) [inline test/packaging/rpm/explicit-target.py:80]: I've fought this before: we are going to need a different solution for this, as it seems the way Python reports the line number changed at 3.8, from last line to first line - we're never going to be able to match it accurately across that boundary.
3.7:
```
============ BEGIN STDERR (len=123)

scons: *** Setting target is not supported for rpm.
File "/tmp/testcmd.1120038.thh9kg5h/SConstruct", line 23, in <module>
```
3.8:
```
============ BEGIN STDERR (len=123)

scons: *** Setting target is not supported for rpm.
File "/tmp/testcmd.1120287.54aen223/SConstruct", line 12, in <module>
```

File: MANIFEST.in [hunk 4]
```diff
@@ -0,0 +1,10 @@
+global-exclude **Tests.py
+
+recursive-include src/engine/SCons/Tool/docbook *
```
Comment (mwichmann) [inline MANIFEST.in:3]: be nice if we could make this a "sub-package".  I can't quite figure out how though.

File: SConstruct [hunk 5]
```diff
@@ -31,100 +31,25 @@ month_year = 'December 2019'
...
 import sys
-import tempfile
-import time
-import socket
 import textwrap
 
-import bootstrap
 
 project = 'scons'
 default_version = '3.1.2'
```
Comment (mwichmann) [inline SConstruct]: Update, maybe?

**Comment by mwichmann** [comment]
Oh the whole it looks good, much as you can tell from a read through of diffs.  git gets confused on file renamings, which didn't make it easier.  Kind of odd that for the command-line scripts, they got to keep the "common part" and import the command-specific part from the new `SCons.Utilities` that has a kind of backwards feel even though I understand.

File: bin/SConsDoc.py [hunk 6]
```diff
@@ -859,45 +851,23 @@ def parseXmlFile(self, fpath):
...
-            loader = importlib._bootstrap_external.SourcelessFileLoader(name, path)
-        else:
-            loader = importlib._bootstrap_external.SourceFileLoader(name, path)
-        # XXX We probably don't need to pass in the loader here.
-        spec = importlib.util.spec_from_file_location(name, path, loader=loader)
-        try:
-            return importlib._bootstrap._load(spec)
-        except ImportError:
-            raise ErrorDuringImport(path, sys.exc_info())
+def importfile(path):
```
Comment (mwichmann) [inline bin/SConsDoc.py:855]: looks like we need an import of `importlib` (preferably at the top), don't know why this hasn't been a problem before - several sider complaints.

File: bin/SConsDoc.py [hunk 7]
```diff
@@ -859,45 +851,23 @@ def parseXmlFile(self, fpath):
...
+    if is_bytecode:
+        loader = importlib._bootstrap_external.SourcelessFileLoader(name, path)
+    else:
+        loader = importlib._bootstrap_external.SourceFileLoader(name, path)
+    # XXX We probably don't need to pass in the loader here.
+    spec = importlib.util.spec_from_file_location(name, path, loader=loader)
+    try:
+        return importlib._bootstrap._load(spec)
+    except ImportError:
+        raise ErrorDuringImport(path, sys.exc_info())
```
Comment (mwichmann) [inline bin/SConsDoc.py]: `ErrorDuringImport` only defined in pydoc, from which this is lifted (see comment - now-removed Py2 version came from there and I just followed along the model)... we'll need to do something here.  Sider complaint. Possibly just `raise` would be ok.

File: site_scons/BuildCommandLine.py [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+                if [l for l in self.git_status_lines if 'modified' in l]:
+                    result = result + '[MODIFIED]'
+                return result
+
+            self.revision = self.git_hash
+
+        self.checkpoint = ARGUMENTS.get('CHECKPOINT', '')
+        if self.checkpoint:
+            if self.checkpoint == 'd':
+                cself.heckpoint = time.strftime('%Y%m%d', time.localtime(time.time()))
```
Comment (mwichmann) [inline site_scons/BuildCommandLine.py]: ???  self.checkpoint, I assume.

File: test/scons-time/run/archive/dir.py [hunk 9]
```diff
@@ -32,6 +32,7 @@
 import TestSCons_time
 
 test = TestSCons_time.TestSCons_time()
+test.verbose_set(1)
```
Comment (mwichmann) [inline test/scons-time/run/archive/dir.py]: needed?

File: setup.cfg (hunk 2 above)
Comment (bdbaddog) [reply setup.cfg:8]: Like Star Trek Next Generation.. that's 30 years ago right? ;)

File: MANIFEST.in (hunk 4 above)
Comment (bdbaddog) [reply MANIFEST.in:3]: Yes. Indeed. I'd like to figure that one out.

**Pull request merged**